lightningd/peer_control.c

```diff
@@ -1200,10 +1203,39 @@ static struct command_result *json_setchannelfee(struct command *cmd,
 		   NULL))
 		return command_param_failed();
 
-	// TODO: write this stuff
+	// find the channel
+	peer = peer_from_json(cmd->ld, buffer, idtok);
+	if (peer) {
+		channel = peer_active_channel(peer);
+		if (!channel)
```

Comment:
You can un-nest the `if`-statements if you move the `!peer` case up, and always return in it. Then the `if (!channel)` case can be unindented.

Reply:
No, we cant do this, same reason because the function returns the error and sets a pointer (the one gcc may complain unset): 

```C
res = command_find_channel(cmd, buffer, idtok, &channel);
```

we could rename `res` to `err` to make it more clear. If it returns NULL it means it did set the `&channel`. Im just using your code here :]